src/mainwindow.cpp

```diff
@@ -4568,6 +4568,7 @@ void MainWindow::on_modList_customContextMenuRequested(const QPoint &pos)
     }
 
     menu->exec(modList->mapToGlobal(pos));
+    delete menu;

```

Comment:
If something's being deleted in the same function as it's created, it's probably better off being a local variable.

Reply:
Right. I don't see that the menu is getting stored anywhere, so it should probably just be created locally. Qt will clean up after itself when it goes out of scope.

Reply:
I guess the main question is whether modListContextMenu results ever get stored where a pointer is necessary. If they're always returned and immediately consumed and then go out of scope, it's not clear to me why the pointer is needed. We might also need to delete the 'sub_menu' item(s) as well, since they were allocated with pointers? Or does Qt clean those up?

Reply:
It does.

Reply:
If a QWidget has a parent, when the parent is deconstructed, the QWidget gets deleted. That means we don't want to stack allocate things we set the parent of unless we know they get deconstructed before the parent (as from @przester's testing, when a child is deconstructed, the parent relationship is undone).